controllers/imageupdateautomation_controller.go

```diff
@@ -216,6 +216,8 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		}
 		pushBranch = ref.Branch
 		tracelog.Info("using push branch from $ref.branch", "branch", pushBranch)
+	} else {
+		failWithError(fmt.Errorf("Invalid reference. Please either specify a push branch, .spec.git.checkout.ref or spec.ref"))
```

Comment:
I think this is the same failure case as `ref.Branch == ""`, so it would be better to put this branch in the same place, to fail with the same error.

Reply:
Agreed - please do see my comment below on adding a return statement too